File: AFMobile/AFAndroid/app/src/main/java/cz/cvut/fel/matyapav/afandroid/enums/SupportedValidations.java
package cz.cvut.fel.matyapav.afandroid.enums;

/**
 * Enumeration of supported validation rules
 *
 *@author Pavel Matyáš (matyapav@fel.cvut.cz).
 *@since 1.0.0..
 */
public enum SupportedValidations {

    REQUIRED("REQUIRED"),
    MAXLENGTH("MAXLENGTH"),
    MINVALUE("MINVALUE"),
    MAXVALUE("MAXVALUE"),
    LESSTHAN("LESSTHAN");

    private String validationType;

    SupportedValidations(String validationType) {
        this.validationType = validationType;
    }

    public String getValidationType() {
        return validationType;
    }
}

File: AFMobile/AFWinPhone/AFWinPhone/enums/SupportedValidations.cs
using System;

namespace AFWinPhone.enums
{
    class SupportedValidations
    {
        public static readonly SupportedValidations REQUIRED = new SupportedValidations("REQUIRED");
        public static readonly SupportedValidations MAXLENGTH = new SupportedValidations("MAXLENGTH");
        public static readonly SupportedValidations MIN = new SupportedValidations("MIN");
        public static readonly SupportedValidations MAX = new SupportedValidations("MAX");
        public static readonly SupportedValidations LESSTHAN = new SupportedValidations("LESSTHAN");
        public static readonly SupportedValidations NUMBER = new SupportedValidations("NUMBER");


        private String validationType;

        SupportedValidations(String validationType)
        {
            this.validationType = validationType;
        }

        public String getValidationType()
        {
            return validationType;
        }
    }
}
